spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/AvroBinaryIterator.java

```diff
@@ -53,17 +56,23 @@ public AvroBinaryIterator(
       List<String> columnsInOrder,
       org.apache.avro.Schema schema,
       ByteString rowsInBytes,
-      Optional<StructType> userProvidedSchema) {
+      Optional<StructType> userProvidedSchema,
+      Optional<BigQueryStorageReadRowsTracer> bigQueryStorageReadRowsTracer) {
     reader = new GenericDatumReader<GenericRecord>(schema);
     this.bqSchema = bqSchema;
     this.columnsInOrder = columnsInOrder;
     in = new DecoderFactory().binaryDecoder(rowsInBytes.toByteArray(), null);
     this.userProvidedSchema = userProvidedSchema;
+    this.bigQueryStorageReadRowsTracer = bigQueryStorageReadRowsTracer;
   }
 
   @Override
   public boolean hasNext() {
     try {
+      // Avro iterator is used in both V1 and V2
+      if (bigQueryStorageReadRowsTracer.isPresent() && in.isEnd()) {
+        bigQueryStorageReadRowsTracer.get().rowsParseFinished(numberOfRowsParsed);
```

Comment:
i think this potentially mingles parsing time with application time.

Reply:
@emkornfield  I see. Which is right place for "rowParseFinished?
Is it after "SchemaConverters.convertToInternalRow" in next() method? Will that cover parsing time for Avro?